thread/common/thrprof.c

```diff
@@ -1025,3 +1030,74 @@ omrthread_get_jvm_cpu_usage_info_error_recovery(void)
 		GLOBAL_UNLOCK_SIMPLE(lib);
 	}
 }
+
+intptr_t
+omrthread_get_thread_times(omrthread_thread_time_t *threadTime)
+{
+#if defined(LINUX)
+	struct rusage rUsage;
+	memset(&rUsage, 0, sizeof(rUsage));
+
+	if (-1 != getrusage(RUSAGE_THREAD, &rUsage)) {
+		threadTime->userTime = (SEC_TO_NANO_CONVERSION_CONSTANT * (int64_t)rUsage.ru_utime.tv_sec)
+			+ (MICRO_TO_NANO_CONVERSION_CONSTANT * (int64_t)rUsage.ru_utime.tv_usec);
+		threadTime->sysTime = (SEC_TO_NANO_CONVERSION_CONSTANT * (int64_t)rUsage.ru_stime.tv_sec)
+			+ (MICRO_TO_NANO_CONVERSION_CONSTANT * (int64_t)rUsage.ru_stime.tv_usec);
+
+		return 0;
+	}
+
+	return -1;
+#elif defined(OMR_OS_WINDOWS) && !defined(BREW) /* defined(LINUX) */
+
```

Comment:
i don't feel such newlines are needed; `#elif/else` are sufficient for separating the implementations.